### PR TITLE
Fix version 1.0 

### PR DESCRIPTION
**Pull request by AntoineJac**
Hello, I have noticed several snags on the new version as previously flagged with Appnexus team.

Here is a PR with many changes and should allow to have a stable version:
- delegate to weak
- run the completion in global not main queue as if a publisher is calling a DispatchQueue.main.sync from the completion it will raise a memory leak 
- unnecessary optionals remove as default value when class is init
- super.init must be done after property initialisation in swift
- fix the timer not bound to main loop (could raise issue in React Native)
- itunesID was not managed by RequestBuilder
- tid must be a random string and not always the same
- default timeout should be 2000ms as stated by prebid org. Also the timeout should be write as publisher may switch it
- removeHBKeywords could be called at start of fetchDemand so we are sure all is ok

Can you please review and push as new version.

Thanks,

Antoine Jacquemin (Rubicon Project)

### CODE REVIEW

**Comment by AntoineJac** [comment]
@ppuviarasu , can you please have a look. Thanks

**Comment by yoalex5** [comment]
Hello @AntoineJac !

Thank you for your PR.

It seems good for me except one thing - it is using `global` queue instead of `main`. 
You are right that if a publisher puts the line `DispatchQueue.main.sync` in a completion handler block it will produce a deadlock situation and crash the application. In general it is not a good practice to call `DispatchQueue.main.sync`. Also, since we make a UI call `self.dfpBanner!.load(self.request)` e.g. it should be done in the UI queue. One more issue to be taken into account is that `Main Thread Checker` show a warning `-[UIView bounds] must be used from main thread only`

screen shot 2019-03-07 at 4 19 54 pm

**Comment by AntoineJac** [comment]
I agree. I have just updated the code. Also 2 others points I forgot to mention:
- Webview to get UA is made in Prebid Class
- Prebid is initialised in appDelegate so we can get the UA on the 1st call

With Swiftlint, I have noticed many others warning (line to longs, force cast, blank or uneeded, trailing whitespeace)... We will need to through it and clean the code but the version of this PR is at lease stable.

**Comment by yoalex5** [comment]
https://github.com/prebid/prebid-mobile-ios/blob/4925a2b6317250b96168c7393a4a4878e9cfd221/src/PrebidMobile/PrebidMobile/Dispatcher.swift#L26 

I agree that a delegate with strong reference  leads to leak memory issue but if to use weak reference in this case the auto-refresh feature will not work. 
It is because there is no reference on BannerAdUnit class, as a result delegate is nil in the fireTimer() method

Screen Shot 2019-03-11 at 8 21 18 PM

I think we should develop more universal solution for proper work with memory and support all necessary features

**Comment by AntoineJac** [comment]
@yoalex5 , I have fixed the delegate issue after our discussion.
@ppuviarasu , could you please push and merge?

Thanks,

File: src/PrebidMobile/PrebidMobile/BidManager.swift [hunk 1]
```diff
@@ -54,7 +46,7 @@ import Foundation
                     if(!Prebid.shared.timeoutUpdated) {
                         let tmax = self.getTmaxRequest(data!)
                         if(tmax > 0) {
-                            Prebid.shared.timeoutMillis = min(demandFetchEndTime - demandFetchStartTime + tmax + 200, 10000)
```
[the remark this answers is not in the document]
Comment (AntoineJac) [reply src/PrebidMobile/PrebidMobile/BidManager.swift]: Hi @ppuviarasu , Prebid timeout should never be over 2000ms, this is the default in PBJS and I would prefer keeping the same timeout here. Especially that DFP is trigger as a callback.

[the remark this answers is not in the document]
Comment (ppuviarasu) [reply src/PrebidMobile/PrebidMobile/BidManager.swift]: Do make the same changes to the Android SDK too

**Pull request merged**